### ISSUE TITLE
Run as normal user

### ISSUE THREAD

**Issue by HarlemSquirrel** [gfduszynski/cm-rgb#4]
I've been able to run this as a normal (non-root) user by creating the following [udev](https://wiki.archlinux.org/index.php/Udev) rules in Arch Linux

/usr/lib/udev/rules.d/99-cm-rgb.rules
```
# cm-rgbi
SUBSYSTEM=="usb", ATTR{idVendor}=="2516", ATTR{idProduct}=="0051", MODE="0660", GROUP="plugdev", TAG+="uaccess", TAG+="udev-acl"
```

Then load the new rules with
```sh
sudo udevadm control --reload-rules
sudo udevadm trigger
```

Maybe this CLI tool could offer a way to create these rules for users with sudo access?

**Comment by gfduszynski** [comment]
I think this would be cool addition.

However I have to figure out how to do this, for it to work on as many distros as possible.
`` GROUP="plugdev" `` is probably not available on every distro.

**Comment by HarlemSquirrel** [comment]
Groups can be created, too :)

**Comment by gfduszynski** [comment]
True :) 
But I think that the best solution would be to have a list of groups for each distro.
This way no user input should be required.

I'll start figuring this out next year.

**Comment by gfduszynski** [comment]
Ok, so I've learned what the ``TAG+="uaccess"`` means :)

According to:
https://enotty.pipebreaker.pl/2012/05/23/linux-automatic-user-acl-management/

If the goal is to have access to the device as current active user this tag is all that is required.
No group should be defined in the rule unless device is to be used from a daemon.

I've even found reference to this group on Arch issue tracker, quote
``Plugdev is an obsolete group to give members of that group full access to removable devices. Permissions on such devices are granted by logind using ACLs these days, so there's no need for plugdev. Any package shipping such rules should be considered buggy.`` -  Jan de Groot 

I guess your rule should work equally well without the plugdev part.

**Comment by gfduszynski** [comment]
This feature is now implemented in ``v0.1.0`` by running ``cm-rgb-cli add-udev-rule``
